src/components/FlexibleForm/Control.js

```diff
@@ -219,37 +219,109 @@ export const Control = memo(props => {
+      initialFields={initialFields}
+      injectedProps={injectedProps}
+      commonSections={commonSections}
+      record={attribs.record}
+      {...cfg}
+    />
+  ));
+
+  const renderMCL = contentData => {
+    const {
+      legend,
+      visibleColumns,
+      columnWidths,
+      columnMapping,
+      mclColumnPaths,
+    } = props;
+
+    const data = isEmpty(mclColumnPaths) ? contentData : contentData.map(row => {
+      let currentRow = {};
+
+      visibleColumns.forEach(col => {
+        const colPaths = mclColumnPaths[col];
+        const fieldName = get(row, colPaths.namePath, col);
+        const fieldValue = get(row, colPaths.valuePath, <stripesComponents.NoValue />);
+
+        currentRow = {
+          ...currentRow,
+          [fieldName]: fieldValue,
+        };
+      });
+
+      return currentRow;
+    });
+
+    let mapping = {};
+
+    Object.keys(columnMapping).forEach(key => {
+      const mapValue = columnMapping[key];
+
+      mapping = {
+        ...mapping,
+        [key]: isTranslationId(mapValue) ? <FormattedMessage id={mapValue} /> : mapValue,
+      };
+    });
+
+    return (
+      <>
+        <div>
+          <strong>{isTranslationId(legend) ? <FormattedMessage id={legend} /> : legend}</strong>
+        </div>
+        <stripesComponents.MultiColumnList
+          contentData={data}
+          visibleColumns={visibleColumns}
+          columnWidths={columnWidths}
+          columnMapping={mapping}
+        />
+        <br />
+      </>
+    );
+  };
+
   const renderDefault = () => {
     const Cmp = !isEditable ? getControl(staticControlType) : getControl(controlType);
 
     if (isFragment) {
       return <Cmp />;
     }
 
+    const {
+      fields,
+      renderStaticAsMCL,
+    } = props;
+    const refTable = get(referenceTables, fields, []);
+
+    if (repeatable && renderStaticAsMCL) {
+      return (
+        <Cmp
+          className={classes}
+          {...attribs}
+        >
+          {renderMCL(refTable)}
+        </Cmp>
+      );
+    }
+
     if (hasChildren(props)) {
       return (
         <Cmp
           className={classes}
           {...attribs}
         >
-          {children.map((cfg, i) => (
-            <Control
-              key={`control-${i}`}
-              repeatableIndex={cfg.repeatable ? i : repeatableIndex}
-              staticNamespace={staticNamespace}
-              editableNamespace={editableNamespace}
-              sectionNamespace={sectionNamespace}
-              intl={intl}
-              styles={styles}
-              referenceTables={referenceTables}
-              setReferenceTables={setReferenceTables}
-              initialFields={initialFields}
-              injectedProps={injectedProps}
-              commonSections={commonSections}
-              record={attribs.record}
-              {...cfg}
-            />
-          ))}
+          {repeatable ? refTable.map(() => renderChildren()) : renderChildren()}
```

Comment:
I see that for the `repeatable` the .map is used both for `refTable` and for [`children`](https://github.com/folio-org/ui-data-import/pull/493/files#diff-6709362c6d184124da81079fe4cef458R222). Please clarify whether it is correct.

In case there are multiple `refTable` and only one child the `repeatableIndex` for each of the child will be `0` because of that.

In case the `index`

Reply:
Yes, all is correct

Reply:
This is done for the cases when the field names and values are not flat an obvious as usual:
```
{
    name: 'notes',
    enabled: true,
    path: 'item.notes',
    value: '',
    subfields: [{
      order: 0,
      path: 'item.notes[]',
      fields: [{
        // order: 0,
        name: 'itemNoteTypeId',
        enabled: true,
        path: 'item.notes[].itemNoteTypeId',
        value: '',
      }, {
        // order: 1,
        name: 'note',
        enabled: true,
        path: 'item.notes[].note',
        value: '',
      }, {
        // order: 2,
        name: 'staffOnly',
        enabled: true,
        path: 'item.notes[].staffOnly',
        value: null,
        booleanFieldAction: 'AS_IS',
      }],
    }],
  }
```

Here reference table for the current repeatable is taken from `subfields` prop, but columns are in its `fields` subsection.